src/main/java/com/onarandombox/MultiverseCore/destination/DestinationFactory.java

```diff
@@ -37,19 +43,103 @@ public DestinationFactory(MultiverseCore plugin) {
     }
 
     /**
-     * Gets a new destination from a string.
+     * Gets a new destination from a string and possibly for a player.
      * Returns a new InvalidDestination if the string could not be parsed.
      *
      * @param destination The destination in string format.
+     * @param player The player.
      *
      * @return A non-null MVDestination
      */
-    public MVDestination getDestination(String destination) {
+    public MVDestination getDestination(String destination, Player player) {
         String idenChar = "";
         if (destination.split(":").length > 1) {
             idenChar = destination.split(":")[0];
         }
 
+        if (idenChar.equals("") && player != null && this.plugin.getMVWorldManager().isMVWorld(destination)) {
+            // handle special case
+            String playerID = player.getUniqueId().toString();
+            File f = new File(this.plugin.getDataFolder(), MultiverseCore.PLAYER_LOCATION_DATA
+                + File.separator + destination + File.separator + playerID + ".yaml");
+            if (f.isFile()) {
+                YamlConfiguration yc = new YamlConfiguration();
+                try {
+                    yc.load(f);
+                } catch (Exception e) {
+                    this.plugin.log(Level.SEVERE, "Failed to load saved location of player '"
+                        + player.getName() + "' (" + playerID + ") in world '" + destination
+                        + "': " + e.getMessage() + ".");
+                    yc = null;
+                    // fall through to use world spawn location
+                }
+                if (yc != null) {
```

Comment:
It's been a while since I've looked at MV's code but I think we have a simpler way to serialize/deserialize location data.